sql/hive/src/main/scala/org/apache/spark/sql/hive/client/ClientWrapper.scala

```diff
@@ -0,0 +1,392 @@
+    }
+    converted
+  }
+
+  private def toInputFormat(name: String) =
+    Class.forName(name).asInstanceOf[Class[_ <: org.apache.hadoop.mapred.InputFormat[_, _]]]
+
+  private def toOutputFormat(name: String) =
+    Class.forName(name)
+      .asInstanceOf[Class[_ <: org.apache.hadoop.hive.ql.io.HiveOutputFormat[_, _]]]
+
+  private def toQlTable(table: HiveTable): Table = {
+    val qlTable = new Table(table.database, table.name)
+
+    qlTable.setFields(table.schema.map(c => new FieldSchema(c.name, c.hiveType, c.comment)))
+    qlTable.setPartCols(
+      table.partitionColumns.map(c => new FieldSchema(c.name, c.hiveType, c.comment)))
+    table.properties.foreach { case (k, v) => qlTable.setProperty(k, v) }
+    table.serdeProperties.foreach { case (k, v) => qlTable.setSerdeParam(k, v) }
+    version match {
+      case hive.v12 =>
+        table.location.map(new URI(_)).foreach(u => qlTable.call[URI, Unit]("setDataLocation", u))
+      case hive.v13 =>
+        table.location
+          .map(new org.apache.hadoop.fs.Path(_))
+          .foreach(qlTable.call[Path, Unit]("setDataLocation", _))
+    }
+    table.inputFormat.map(toInputFormat).foreach(qlTable.setInputFormatClass)
+    table.outputFormat.map(toOutputFormat).foreach(qlTable.setOutputFormatClass)
+    table.serde.foreach(qlTable.setSerializationLib)
+
+    qlTable
+  }
+
+  def createTable(table: HiveTable): Unit = withClassLoader {
+    val qlTable = toQlTable(table)
+    client.createTable(qlTable)
+  }
+
+  def alterTable(table: HiveTable): Unit = withClassLoader {
+    val qlTable = toQlTable(table)
+    client.alterTable(table.qualifiedName, qlTable)
+  }
+
+  def getTables(dbName: String): Seq[String] = withClassLoader {
+    client.getAllTables(dbName).toSeq
+  }
+
+  def getAllPartitions(hTable: HiveTable): Seq[HivePartition] = withClassLoader {
+    val qlTable = toQlTable(hTable)
+    val qlPartitions = version match {
+      case hive.v12 => client.call[Table, Set[Partition]]("getAllPartitionsForPruner", qlTable)
+      case hive.v13 => client.call[Table, Set[Partition]]("getAllPartitionsOf", qlTable)
+    }
+    qlPartitions.map(_.getTPartition).map { p =>
+      HivePartition(
+        values = Option(p.getValues).map(_.toSeq).getOrElse(Seq.empty),
+        storage = HiveStorageDescriptor(
+          location = p.getSd.getLocation,
+          inputFormat = p.getSd.getInputFormat,
+          outputFormat = p.getSd.getOutputFormat,
+          serde = p.getSd.getSerdeInfo.getSerializationLib))
+    }.toSeq
+  }
+
+  def listTables(dbName: String): Seq[String] = withClassLoader {
+    client.getAllTables
+  }
+
+  /**
+   * Runs the specified SQL query using Hive.
+   */
+  def runSqlHive(sql: String): Seq[String] = {
+    val maxResults = 100000
+    val results = runHive(sql, maxResults)
+    // It is very confusing when you only get back some of the results...
+    if (results.size == maxResults) sys.error("RESULTS POSSIBLY TRUNCATED")
```

Comment:
This is is actually mostly copied from Shark :) which would silently truncate the results.  Its only used when generating golden answer files and I never actually expect a test query to return more than 100,000 results.  If it does however I want an error instead of being confused again.